libs/wingsdk/src/target-sim/container.inflight.ts

```diff
@@ -12,27 +12,23 @@ import { Util } from "../util";
 export class Container implements IContainerClient, ISimulatorResourceInstance {
   private readonly imageTag: string;
   private readonly containerName: string;
+  private _context: ISimulatorContext | undefined;
 
-  public constructor(
-    private readonly props: ContainerSchema["props"],
-    private readonly context: ISimulatorContext
-  ) {
+  public constructor(private readonly props: ContainerSchema["props"]) {
     this.imageTag = props.imageTag;
 
     this.containerName = `wing-container-${Util.ulid()}`;
   }
 
-  private log(message: string) {
-    this.context.addTrace({
-      data: { message },
-      sourcePath: this.context.resourcePath,
-      sourceType: "container",
-      timestamp: new Date().toISOString(),
-      type: TraceType.LOG,
-    });
+  private get context(): ISimulatorContext {
```

Comment:
Kind of feels like there might be some value in creating a base class for SimulatorResourceInstance that allows reusing `context`, `log`, etc.